yjit/src/backend/arm64/mod.rs

```diff
@@ -61,10 +61,7 @@ impl Assembler
     /// Get the list of registers from which we can allocate on this platform
     pub fn get_alloc_regs() -> Vec<Reg>
     {
-        vec![
-            X12_REG,
-            X13_REG
-        ]
+        vec![C_RET_REG, X12_REG]
```

Comment:
We did do this on the X86 side. It's explicitly a part of the vec here: https://github.com/Shopify/ruby/blob/1f85630e00d1f4fb0744e6c5b94fe0e607244a2e/yjit/src/backend/x86_64/mod.rs#L74. This is necessary because we explicitly try to take that register from the pool here: https://github.com/Shopify/ruby/blob/1f85630e00d1f4fb0744e6c5b94fe0e607244a2e/yjit/src/backend/ir.rs#L684. 

I'm happy to change this if you want to, but I was trying to match what X86 did here.

Reply:
Ahhh right because we assume that call kills everything else that's live. Ok makes sense 👍